src/appleseed.python/bindblenderprogressivetilecallback.cpp

```diff
@@ -172,20 +174,106 @@ namespace
                     glBindTexture(GL_TEXTURE_2D, 0);
                     m_updated_buffer = false;
                 }
-
-                glColor4f(1.0f, 1.0f, 1.0f, 1.0f);
-
-                glEnable(GL_TEXTURE_2D);
-                glBindTexture(GL_TEXTURE_2D, m_texture_id);
-                glBegin(GL_QUADS);
-                    glTexCoord2f(0.0f, 1.0f); glVertex3f(x    ,  y    , 0.0f);
-                    glTexCoord2f(1.0f, 1.0f); glVertex3f(x + w,  y    , 0.0f);
-                    glTexCoord2f(1.0f, 0.0f); glVertex3f(x + w,  y + h, 0.0f);
-                    glTexCoord2f(0.0f, 0.0f); glVertex3f(x    ,  y + h, 0.0f);
-                glEnd();
-                glBindTexture(GL_TEXTURE_2D, 0);
-                glDisable(GL_TEXTURE_2D);
-            }
+                if(!m_updated_data_buffer)
+                {
+                  const char *vertex_shader_source  = "#version 330 core\n"
+                                                      "layout (location = 0) in vec3 pos;\n"
+                                                      "layout (location = 1) in vec3 tex_coords;\n"
+                                                      "out texture_coords;\n"
+                                                      "void main()\n"
+                                                      "{\n"
+                                                      "   gl_Position = vec4(pos, 1.0);\n"
+                                                      "   texture_coords = vec3(tex_coords);\n
+                                                      "}\0";
+          
+                  const char *fragment_shader_source = "#version 330 core\n"
+                                                       "out vec4 tex;\n"
+                                                       "in vec3 texture_coords;\n"
+                                                       "unform sampler2D texture;\n"
+                                                       "void main()\n"
+                                                       "{\n"
+                                                       "   tex = vec4(texture, texture_coords);\n"
+                                                       "}\n\0";
+                
+                
+                  GLuint vertex_shader_id = glCreateShader(GL_VERTEX_SHADER);
+                  glShaderSource(vertex_shader_id, 1, &vertex_shader_source, nullptr);
+                  glCompileShader(vertex_shader_id);
+                  // Check for vertex_shader compile errors.
+                  int success;
+                  char infoLog[512];
+                  glGetShaderiv(vertex_shader_id, GL_COMPILE_STATUS, &success);
+                  if (!success)
+                  {
+                      glGetShaderInfoLog(vertex_shader_id, 512, nullptr, infoLog);
+                      std::cout << "ERROR::SHADER::VERTEX::COMPILATION_FAILED\n" << infoLog << std::endl;
```

Comment:
Don't write to `std::cout`, that's meaningless from the point of view of an integration (i.e. our Blender plugin).

Reply:
Should I leave it without any error message or is there any method to show error output from the point of view of an integration